Sources/Logging/Logging.swift

```diff
@@ -43,13 +43,19 @@ public struct Logger {
     /// An identifier of the creator of this `Logger`.
     public let label: String
 
+    /// The metadata provider this logger was created with.
+    /// If no provider was configured, this will return a ``Logger/MetadataProvider/noop`` no-op metadata provider.
+    public var metadataProvider: Logger.MetadataProvider {
+        return self.handler.metadataProvider
+    }
+
     internal init(label: String, _ handler: LogHandler) {
         self.label = label
         self.handler = handler
     }
 }
 
-extension Logger {
+public extension Logger {
```

Comment:
FYI, we can change SwiftFormat's behavior with `--extensionacl on-declarations`

Reply:
I'll add it to `.swiftformat`, that';s what we want 👏

Reply:
Heh so that's part of the problem; the version we use is too old to have this flag;
Other sswg projects use newer formatters, so it is hard to keep it all aligned.

We should bump the formatter version everywhere and fix this mess this way, and by adding more flags. I'll undo bad formats now and avoid reformatting automatically heh.

Reply:
ticketified https://github.com/apple/swift-log/issues/239